### PR TITLE
[5.1] Add array dig function

### PR DESCRIPTION
**Pull request by haleksandre**
Add Arr::dig($array) as well as the array_dig($array) helper.

Array dig convert a "dot" notion into a deep nested array.  Works similar to Arr::set() but retains the "dot" notion array value.

### CODE REVIEW

File: src/Illuminate/Support/Arr.php [hunk 1]
```diff
@@ -67,6 +67,23 @@ public static function collapse($array)
...
+     *
+     * @param  array  $array
+     * @return array
+     */
+    public static function dig($array)
+    {
+        $results = [];
+
+        foreach ($array as $key => $value) {
+            array_set($results, $key, $value);
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/Arr.php]: don't call the helper function, call the static  method on this class

Comment (GrahamCampbell) [inline src/Illuminate/Support/Arr.php]: `static::set(`

File: src/Illuminate/Support/helpers.php [hunk 2]
```diff
@@ -71,6 +71,19 @@ function array_collapse($array)
     }
 }
 
+if (! function_exists('array_dig')) {
+    /**
+     * Create a deep nested array using "dot" notation.
+     *
+     * @param  array   $array
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/helpers.php]: cs

**Comment by taylorotwell** [comment]
Will stick to letting developers define this in their own applications.

**Pull request closed without merging**